boards/metro_m4/Cargo.toml

```diff
@@ -56,9 +58,14 @@ debug = true
 lto = true
 
 [profile.release]
+debug = true
```

Comment:
This bloats file size by a lot.

Reply:
On disk filesize, but literally none of that should pass through to the actual flashed binary

Reply:
With and without

```
jacob@j:~/Downloads/atsamd/boards/samd11_bare (master)$ cargo size --example adc --release --features=unproven
    Finished release [optimized] target(s) in 0.77s
   text    data     bss     dec     hex filename
   5548       0      16    5564    15bc adc
jacob@j:~/Downloads/atsamd/boards/samd11_bare (master)$ 
```

```
jacob@j:~/Downloads/atsamd/boards/samd11_bare (master)$ cargo size --example adc --release --features=unproven
    Finished release [optimized] target(s) in 0.77s
   text    data     bss     dec     hex filename
   5548       0      16    5564    15bc adc
jacob@j:~/Downloads/atsamd/boards/samd11_bare (master)$ 
```


However, all the format and rtt buffers do add a bunch
```
jacob@j:~/Downloads/atsamd/boards/samd11_bare (probe-rs)$ cargo size --example adc --release --features=unproven
    Finished release [optimized] target(s) in 0.01s
   text    data     bss     dec     hex filename
   6240       0    1088    7328    1ca0 adc
```

Reply:
And note, im going to delete the m4 commit unless you want otherwise since it has a bootloader. It was mostly just to give you a target to inspect and play with.